opal/util/bipartite_graph.c

```diff
@@ -817,7 +817,9 @@ static int min_cost_flow_ssp(opal_bp_graph_t *gx, int **flow_out)
     return err;
 
 out_error:
-    free(*flow_out);
+    if (NULL != flow_out && NULL != *flow_out) {
```

Comment:
```
if (NULL != flow_out) {
    free(*flow_out);
}
```

should be sufficient.  Freeing a NULL pointer is legal in C99.

Reply:
I did that initially, but still got a warning.  Your comment made me look deeper, though, and I think I found the root cause:

```patch
diff --git a/opal/util/malloc.h b/opal/util/malloc.h
index 5ff79c57a3..c33779476e 100644
--- a/opal/util/malloc.h
+++ b/opal/util/malloc.h
@@ -127,7 +127,7 @@ OPAL_DECLSPEC void *opal_realloc(void *ptr, size_t size, const char *file, int l
  * to configure (or by default if you're building in a SVN
  * checkout).
  */
-OPAL_DECLSPEC void opal_free(void *addr, const char *file, int line) __opal_attribute_nonnull__(1);
+OPAL_DECLSPEC void opal_free(void *addr, const char *file, int line)
 
 /**
  * Used to set the debug level for malloc debug.
```

Since `free(NULL)` is legal in C99, we shouldn't put a compiler directive in our debugging-level `free()` wrapper mandating that the 1st argument is non-NULL.